central/cluster/datastore/datastore_impl_postgres_test.go

```diff
@@ -31,29 +62,754 @@ func TestClusterDataStoreWithPostgres(t *testing.T) {
 
 	s.ctx = sac.WithAllAccess(context.Background())
-
 	s.db = pgtest.ForT(s.T())
+	clusterDBStore := clusterPostgresStore.New(s.db)
+	clusterHealthDBStore := clusterHealthPostgresStore.New(s.db)
+	nodeStore := nodeDataStore.GetTestPostgresDataStore(s.T(), s.db)
+	netFlowStore, err := netFlowsDataStore.GetTestPostgresClusterDataStore(s.T(), s.db)
+	s.NoError(err)
+	netEntityStore, err := netEntityDataStore.GetTestPostgresDataStore(s.T(), s.db)
+	s.NoError(err)
+	networkBaselineM, err := networkBaselineManager.GetTestPostgresManager(s.T(), s.db)
+	s.NoError(err)
+	clusterCVEStore, err := clusterCVEDataStore.GetTestPostgresDataStore(s.T(), s.db)
+	s.NoError(err)
+	hashStore, err := datastore.GetTestPostgresDataStore(s.T(), s.db)
+	s.NoError(err)
+	sensorCnxMgr := connection.NewManager(hashManager.NewManager(hashStore))
+	clusterRanker := ranking.ClusterRanker()
+	compliancePruner := compliancePruning.GetTestPruner(s.T(), s.db)
+	s.nsDatastore, err = namespace.GetTestPostgresDataStore(s.T(), s.db.DB)
+	s.NoError(err)
+	s.alertDatastore, err = alertDatastore.GetTestPostgresDataStore(s.T(), s.db.DB)
+	s.NoError(err)
+	s.deploymentDatastore, err = deploymentDatastore.GetTestPostgresDataStore(s.T(), s.db.DB)
+	s.NoError(err)
+	s.podDatastore, err = podDatastore.GetTestPostgresDataStore(s.T(), s.db.DB)
+	s.NoError(err)
+	s.secretDatastore, err = secretDataStore.GetTestPostgresDataStore(s.T(), s.db.DB)
+	s.NoError(err)
+	s.serviceAccountDatastore, err = serviceAccountDataStore.GetTestPostgresDataStore(s.T(), s.db.DB)
+	s.NoError(err)
+	s.roleDatastore = k8sRoleDataStore.GetTestPostgresDataStore(s.T(), s.db.DB)
+	s.roleBindingDatastore = k8sRoleBindingDataStore.GetTestPostgresDataStore(s.T(), s.db.DB)
+	s.imageIntegrationDatastore, err = imageIntegrationDataStore.GetTestPostgresDataStore(s.T(), s.db.DB)
+	s.NoError(err)
+	s.clusterDatastore, err = New(clusterDBStore, clusterHealthDBStore, clusterCVEStore,
+		s.alertDatastore, s.imageIntegrationDatastore, s.nsDatastore, s.deploymentDatastore,
+		nodeStore, s.podDatastore, s.secretDatastore, netFlowStore, netEntityStore,
+		s.serviceAccountDatastore, s.roleDatastore, s.roleBindingDatastore, sensorCnxMgr, nil,
+		clusterRanker, networkBaselineM, compliancePruner)
+	s.NoError(err)
+}
+
+func (s *ClusterPostgresDataStoreTestSuite) TearDownTest() {
+	s.db.Teardown(s.T())
+}
+
+// Test that when we try to remove a cluster that does not exist, we return an error.
+func (s *ClusterPostgresDataStoreTestSuite) TestHandlesClusterDoesNotExist() {
+	ctx := sac.WithAllAccess(context.Background())
+
+	err := s.clusterDatastore.RemoveCluster(ctx, fakeClusterID, nil)
+	// Expect that there is an error because our cluster does not exist.
+	s.Error(err)
+}
+
+func (s *ClusterPostgresDataStoreTestSuite) TestRemoveCluster() {
+	ctx := sac.WithAllAccess(context.Background())
+
+	testCluster := &storage.Cluster{Name: fixtureconsts.ClusterName1, MainImage: mainImage, CentralApiEndpoint: centralEndpoint}
```

Comment:
nit:  readability concern.  I think I would add some new lines periodically.  Like after each logical block.

For example:
```
	s.NoError(clusterAddErr)

	testDeployment := &storage.Deployment{Id: fixtureconsts.Deployment1, ClusterId: clusterId, ClusterName: testCluster.Name}
```